internal/utils.go

```diff
@@ -0,0 +1,66 @@
+package internal
+
+import (
+	"bytes"
+	"fmt"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+)
+
+func ReadStdin() (string, error) {
+	info, err := os.Stdin.Stat()
+	if err != nil {
+		return "", err
+	}
+
+	if (info.Mode() & os.ModeCharDevice) != 0 {
+		return "", nil
+	}
+
+	buf := &bytes.Buffer{}
+	_, err = buf.ReadFrom(os.Stdin)
+	if err != nil {
+		return "", fmt.Errorf("reading from stdin: %w", err)
+	}
+
+	return buf.String(), nil
+}
+
+func validateAddress(addr string) bool {
+
+	// URL validation
+	if _, err := url.ParseRequestURI(addr); err == nil {
```

Comment:
Please move `err == nil` to the next line. We are not using this patter in the codebase.